### PR TITLE
fix: repo upgrade

### PR DESCRIPTION
**Pull request by nomego**
- drop unused files
  - analysis.json
  - index.html
  - polymer.json
  - wct.conf.json
- switch to karm
  - add karma problem matcher
- switch github workflow to setup-node eslint problem matcher
- update codeclimate conf
- update vscode settings
- add es-dev-server
- upgrade deps
- convert tests to karma
- add tests for found bugs
- fix bug with selectedInstance never getting set
- add parallelDataRequests config prop

### CODE REVIEW

**Comment by nomego** [comment]
`parallelDataRequests` behavior configuration added in this PR too.

File: cosmoz-data-nav.js [hunk 1]
```diff
@@ -751,11 +757,11 @@ class CosmozDataNav extends translatable(mixinBehaviors([IronResizableBehavior],
 	}
 
 	_getInstance(selectedElement) {
-		if (selectedElement == null || selectedElement.children.length < 2) {
+		if (selectedElement == null) {
 			return;
 		}
 		// index 0 is incomplete element
```
Comment (Wurper) [inline cosmoz-data-nav.js]: Is this comment still true/relevant?

**Review by Wurper — APPROVED** [review]

File: test/helpers/utils.js [hunk 2]
```diff
@@ -1,9 +1,66 @@
...
+					type: Boolean
+				},
+				nextDisabled: {
+					type: Boolean
+				},
+				notifyProp: {
+					type: Boolean,
+					notify: true,
+					value: false
+				}
```
Comment (cristinecula) [inline test/helpers/utils.js]: What is this used for?

File: test/bugs.test.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+	test('https://github.com/Neovici/cosmoz-data-nav/issues/117', async () => {
+		const nav = await fixture(defaultsFixture);
+		nav.parallelDataRequests = true;
+		try {
+			await oneRequestPerItem(nav);
+		} catch (e) {
+			assert.equal(e.message, 'requests for id 1: expected 3 to equal 1');
+			return;
+		}
```
Comment (cristinecula) [inline test/bugs.test.js:116]: Asserting an assert error message is very smelly and very hard to understand it's purpose. Try expressing the assertion differently.

Also, tests do not need to be so DRY. I would refactor `oneRequestPerItem` to a beforeTest hook. You can create a different suite so you don't affect the other test.

**Review by cristinecula — CHANGES_REQUESTED** [review]
Data-nav is already doing waaay too much as it is, so I'm very reluctant to complicate it's behavior even more with the parallel-data-requests flag. Is it really an issue that data-nav requests the same item multiple times? Shouldn't it be the responsibility of the layer above data-nav that deduplicates the requests?

**Review by cristinecula — APPROVED** [review]
As discussed on Slack, this change is actually making data-nav less opinionated. :+1:

**Review by Wurper — APPROVED** [review]

**Review by cristinecula — APPROVED** [review]

**Pull request merged**